starter-code/src/viking.js

```diff
@@ -1,11 +1,440 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor (health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength;
+    }
+    receiveDamage(damage){
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor (name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    } 
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return (`${this.name} has received ${damage} points of damage`);
+        }
+        else {
+            return(`${this.name} has died in act of combat`);
+        }
+    }
+    battleCry(){
+        return(`Odin Owns You All!`);
+    }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength){
+        super(health, strenght);
+    }
+    receiveDamage(damage){
+        this.health -= damage;
+        if (this.health > 0) {
+            return (`A Saxon has received ${damage} points of damage`);
+        }
+        else {
+            return (`A Saxon has died in act of combat`);
+        }
+    }
+
+
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        let vikingArmy = "";
+        let saxonArmy = "";
+
+        addViking(Viking){
+            vikingArmy += Viking;
+        }
+        addSaxon(Saxon){
+            saxonArmy += Saxon;
+        }
+
+    }
```

Comment:
Alors ici, plusieurs choses :)

1. Dans un constructor, quand on déclare une variable propre à une classe, comme ici vikingArmy et saxonArmy doivent l'être, la syntaxe est : `this.vikingArmy` et `this.saxonArmy`;

2. Vos deux variables d'armées sont des strings, mais du coup il va être compliqué de leur faire contenir des objets, étant donné que tout ce que peut contenir un string est... des caractères :) Du coup, il aurait été plus intéressant de les déclarer comme des arrays, pour pouvoir par la suite leur pousser les objets `viking` et `saxon` : `this.vikingArmy.push(viking)`;

3. les méthodes d'une classe ne doivent pas être déclarées dans le constructor, mais après. Vous devriez donc plutôt avoir :
```javascript
    constructor() {
        let vikingArmy = "";
        let saxonArmy = "";
    }

    addViking(Viking){
        vikingArmy += Viking;
    }
    addSaxon(Saxon){
        saxonArmy += Saxon;
    }
```